pkg/cmd/pipeline/list_test.go

```diff
@@ -3,38 +3,205 @@ package pipeline
+		"",
+	}
+
+	text := strings.Join(expected, "\n")
+	if d := cmp.Diff(text, output); d != "" {
 		t.Errorf("Unexpected output mismatch: %s", d)
 	}
 }
+
+func TestPipelinesList_with_single_run(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+
+	cs, _ := test.SeedTestData(test.Data{
+		Pipelines: []*v1alpha1.Pipeline{
+			tb.Pipeline("pipeline", "ns",
+				// created  5 minutes back
+				testutil.PipelineCreationTimestamp(clock.Now().Add(-5*time.Minute)),
+			),
+		},
+		PipelineRuns: []*v1alpha1.PipelineRun{
+
+			tb.PipelineRun("pipeline-run-1", "ns",
+				tb.PipelineRunLabel("tekton.dev/pipeline", "pipeline"),
+				tb.PipelineRunSpec("pipeline"),
+				tb.PipelineRunStatus(
+					tb.PipelineRunStatusCondition(apis.Condition{
+						Status: corev1.ConditionTrue,
+						Reason: resources.ReasonSucceeded,
+					}),
+					// pipeline run starts now
+					tb.PipelineRunStartTime(clock.Now()),
+					// takes 10 minutes to complete
+					testutil.PipelineRunCompletionTime(clock.Now().Add(10*time.Minute)),
+				),
+			),
+		},
+	})
+
+	p := &testutil.TestParams{Client: cs.Pipeline, Clock: clock}
+	pipeline := Command(p)
+
+	// -5 : pipeline created
+	//  0 : pipeline run - 1 started
+	// 10 : pipeline run - 1 finished
+	// 15 : <<< now run pipeline ls << - advance clock to this point
+
+	clock.Advance(15 * time.Minute)
+	got, err := testutil.ExecuteCommand(pipeline, "list", "-n", "ns")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	expected := []string{
+		"NAME       AGE              LAST RUN         STARTED          DURATION     STATUS",
+		"pipeline   20 minutes ago   pipeline-run-1   15 minutes ago   10 minutes   Succeeded",
+		"",
+	}
+
+	text := strings.Join(expected, "\n")
+	if d := cmp.Diff(text, got); d != "" {
+		t.Errorf("Unexpected output mismatch: \n%s\n", d)
+	}
+}
+func TestPipelinesList_latest_run(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+	var (
+		pipelineCreated = clock.Now().Add(-5 * time.Minute)
+		runDuration     = 5 * time.Minute
+
+		firstRunStarted   = clock.Now().Add(10 * time.Minute)
+		firstRunCompleted = firstRunStarted.Add(runDuration)
+
+		secondRunStarted   = firstRunCompleted.Add(5 * time.Minute)
+		secondRunCompleted = secondRunStarted.Add(runDuration)
+	)
+
+	cs, _ := test.SeedTestData(test.Data{
+		Pipelines: []*v1alpha1.Pipeline{
+			tb.Pipeline("pipeline", "ns",
+				// created  5 minutes back
+				testutil.PipelineCreationTimestamp(pipelineCreated),
+			),
+		},
+		PipelineRuns: []*v1alpha1.PipelineRun{
+
+			tb.PipelineRun("pipeline-run-1", "ns",
+				tb.PipelineRunLabel("tekton.dev/pipeline", "pipeline"),
+				tb.PipelineRunSpec("pipeline"),
+				tb.PipelineRunStatus(
+					tb.PipelineRunStatusCondition(apis.Condition{
+						Status: corev1.ConditionTrue,
+						Reason: resources.ReasonSucceeded,
+					}),
+					tb.PipelineRunStartTime(firstRunStarted),
+					testutil.PipelineRunCompletionTime(firstRunCompleted),
+				),
+			),
+			tb.PipelineRun("pipeline-run-2", "ns",
```

Comment:
This PipelineRun needs a creation date too :angel: 
(and if you put a creation date "before" `pipeline-run-1` you'll see it succeeds although `pipeline-run-1` is the *last run* — and thus should have failed)

Reply:
See https://github.com/tektoncd/cli/compare/master...vdemeester:tasks#diff-acd75ed12ce8066b2e1a5f9a1f25c427R150

Reply:
yes, I found that when I fixed the `listPipelineDetails` function. Do you think sorting by `Status.StartTime` instead of creationTimestamp better / expected behaviour ?

Reply:
Fixed ...